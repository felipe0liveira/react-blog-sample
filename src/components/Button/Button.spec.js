import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Button } from '.'

describe('<Button />', () => {
  it('should render the button with a text "Load more" as it\'s content', () => {
    render(<Button text='Load more' />)

    const button = screen.getByRole('button', { name: /load more/i })
    expect(button).toBeInTheDocument()
  })

  it('should call function on button click', () => {
    const fn = jest.fn()

    render(<Button text='Load more' onClick={fn} />)

    const button = screen.getByRole('button', { name: /load more/i })

    // Could be fireEvent/userEvent
    userEvent.click(button)
    expect(fn).toHaveBeenCalledTimes(1)
  })

  it('should be enabled when disabled prop is false', () => {
    render(<Button text='Load more' disabled={false} />)

    const button = screen.getByRole('button', { name: /load more/i })
    expect(button).toBeEnabled()
  })

  it('should be enabled when disabled prop is not set', () => {
    render(<Button text='Load more' />)

    const button = screen.getByRole('button', { name: /load more/i })
    expect(button).toBeEnabled()
  })

  it('should ,atch snapshot', () => {
    render(<Button text='Load more' />)

    const button = screen.getByRole('button', { name: /load more/i })
    expect(button).toBeEnabled()
  })

  it('should be disabled when disabled prop is true', () => {
    const fn = jest.fn()

    const { container } = render(<Button text='Load more' onClick={fn} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
