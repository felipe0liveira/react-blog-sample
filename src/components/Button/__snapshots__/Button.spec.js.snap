// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be disabled when disabled prop is true 1`] = `
<button
  class="button"
>
  Load more
</button>
`;
